### PR TITLE
Update midterm.md

### PR DESCRIPTION
**Pull request by VeeDoubleYuh**

### CODE REVIEW

File: quizzes/midterm.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-1. Circle all 5 of the problems with the code below.
+1. Circle all 5 of the problems with in code below.
```
Comment (seanculver) [inline quizzes/midterm.md:1]: 8/10

File: quizzes/midterm.md [hunk 2]
```diff
@@ -15,6 +15,12 @@
...
 	monkey.eat_banana
 	```
+	```
+	#Cannot inherit from itself
+	#Method must be called initialize
+	#my_vars should be equal to @my_vars
+	#Cannot call private method/Method should be public
+	```
 
 1. What is returned from the statement below?
```
Comment (seanculver) [inline quizzes/midterm.md:25]: 0/10. Too hard of a question.

File: quizzes/midterm.md [hunk 3]
```diff
@@ -23,6 +29,9 @@
 
 	thoughts[2]
 	```
+	```
+	#index access error
+	```
 
 1. What is the result of this `fetch`?
```
Comment (seanculver) [inline quizzes/midterm.md:36]: 10/10

File: quizzes/midterm.md [hunk 4]
```diff
@@ -30,17 +39,20 @@
 	my_hash = { my_key: 'value', another_key: 'another value' }
 	my_hash.fetch(:not_a_key_yet, 'default value')  # => ??????
 	```
+	```
+	#default value
+	```
 
 1. Give an example of a good variable name, and a bad variable name.
```
Comment (seanculver) [inline quizzes/midterm.md:46]: 10/10

File: quizzes/midterm.md [hunk 5]
```diff
@@ -30,17 +39,20 @@
...
-
-
-
-
+	```
+	this_is_a_variable #Good
+	whatTheh3llAMi #Bad
+	```
 
 1. What is the difference between `1..10` and `1...10`?
```
Comment (seanculver) [inline quizzes/midterm.md:52]: 10/10

File: quizzes/midterm.md [hunk 6]
```diff
@@ -51,14 +63,26 @@
 
 	ant_eater
```
Comment (seanculver) [inline quizzes/midterm.md:64]: 0/10 - raises exception.

File: quizzes/midterm.md [hunk 7]
```diff
@@ -51,14 +63,26 @@
...
 	```
+	```
+	#ants is now equal to one less than it was before the method was called.
+	```
 
 1. Name one thing you would fix with the arguments of this method. (see below)
```
Comment (seanculver) [inline quizzes/midterm.md:70]: 10/10

File: quizzes/midterm.md [hunk 8]
```diff
@@ -51,14 +63,26 @@
...
+	```
+	#I would reverse the order of the parameters, as the default looks better on the right...
+	```
 1. Now use the method to eat some humans! (see below)
```
Comment (seanculver) [inline quizzes/midterm.md:74]: 10/10

File: quizzes/midterm.md [hunk 9]
```diff
@@ -51,14 +63,26 @@
...
+	```
+	eat_humans(12, 3)
+	```
 1. write down how many Humans did you eat? (see below)
```
Comment (seanculver) [inline quizzes/midterm.md:78]: 5/10

File: quizzes/midterm.md [hunk 10]
```diff
@@ -66,22 +90,31 @@
 	end
 	```
 
-
 1. Demonstrate using multiple logical operators (||, &&)
```
Comment (seanculver) [inline quizzes/midterm.md:93]: 10/10

File: quizzes/midterm.md [hunk 11]
```diff
@@ -66,22 +90,31 @@
...
-
+	```ruby
+	if (1 > 2 || 5 > 3 && true)
+		doThis
+	end
+	```
 
 1. What is the result of calling sort?
```
Comment (seanculver) [inline quizzes/midterm.md:100]: 10/10 i'm ignoring the typo

File: quizzes/midterm.md [hunk 12]
```diff
@@ -66,22 +90,31 @@
...
 
 	```ruby
 	["one fish", "two fish", "red fish", "blue fish"].sort
 	```
-
+	```ruby
+	#returns ["blue fish", "one fish", "red fish", [two fish"]
+	```
 
 1. Write a hash that has the following keys :name, :super_power, :weakness
```
Comment (seanculver) [inline quizzes/midterm.md:109]: 10/10

File: quizzes/midterm.md [hunk 13]
```diff
@@ -66,22 +90,31 @@
...
-
+	```ruby
+	{name: "Superman", super_power: "Being super", weakness: "Kryptonite"}
+	```
 
 1. T/F You can set a default value for a Hash
```
Comment (seanculver) [inline quizzes/midterm.md:114]: 10/10

File: quizzes/midterm.md [hunk 14]
```diff
@@ -66,22 +90,31 @@
...
-
+	```
+	True
+	```
 
 
 1. What is the resulting hash?
```
Comment (seanculver) [inline quizzes/midterm.md:120]: 10/10

File: quizzes/midterm.md [hunk 15]
```diff
@@ -90,18 +123,29 @@
 	my_hash = { "a" => 100, "b" => 200, "c" => 300 }
 	my_hash.select {|key,value| key < "c"}
 	```
+	```ruby
+	{"a" => 100, "b" => 200}
+	```
 
 1. Write code that iterates through the numbers array and out`puts` the `KEY` for `each` of the values in the array.
```
Comment (seanculver) [inline quizzes/midterm.md:130]: 5/10 - each_with_index

File: quizzes/midterm.md [hunk 16]
```diff
@@ -90,18 +123,29 @@
...
 	```ruby
 	numbers = ['one', 'two', 'three']
 	```
+	```ruby
+	numbers.print_each_key do
+		puts numbers
+	end
+	```
 
 1. Write code that `maps` the fruits array to a new array where all the values are `reverse`d. Eg. "apples" = "selppa"
```
Comment (seanculver) [inline quizzes/midterm.md:141]: 10/10 no need for the bang!

File: quizzes/midterm.md [hunk 17]
```diff
@@ -134,6 +178,9 @@
 	```ruby
 	"Always rzmzmbzr you arz absolutzly uniquz, likz zvzryonz zlsz"
 	```
+	```ruby
```
Comment (seanculver) [inline quizzes/midterm.md:181]: 10/10

File: quizzes/midterm.md [hunk 18]
```diff
@@ -134,6 +178,9 @@
...
+	"Always rzmzmbzr you arz absolutzly uniquz, likz zvzryonz zlsz".gsub("z", "e") #Idk...
+	```
 
 1. What is the resulting Hash. Please fully format the hash with proper `{}` and use `symbols`
```
Comment (seanculver) [inline quizzes/midterm.md:185]: 10/10

File: quizzes/midterm.md [hunk 19]
```diff
@@ -142,13 +189,30 @@
 	  other_hash = { new_key: "new value", new_another: "I will overwrite or will I?"}
 	  my_hash.merge(other_hash)
 	```
-
+	```ruby
+	{my_key: 'value', another_key: 'another value', new_key: "new value", new_another: "I will overwrite or will I?"}
+	```
 
 1. What is the first thing you should do with `Legacy Code?`. And no, you can't throw it out.
```
Comment (seanculver) [inline quizzes/midterm.md:196]: 5/10 Thats a really good answer, what I was looking for was test it.

File: quizzes/midterm.md [hunk 20]
```diff
@@ -142,13 +189,30 @@
...
+	```
+	#Figure out what it does/what its supposed to do.
+	```
 
 1. Who is the creator of Ruby?
```
Comment (seanculver) [inline quizzes/midterm.md:201]: 10/10

File: quizzes/midterm.md [hunk 21]
```diff
@@ -142,13 +189,30 @@
...
+	```
+	#Some japanese dude that has the nickname Matz
+	```
 
 1. Write a class that uses a Global, Instance, Class, and local variable. Don't worry about what it does, just properly use the variables.
```
Comment (seanculver) [inline quizzes/midterm.md:206]: 10/10

File: quizzes/midterm.md [hunk 22]
```diff
@@ -142,13 +189,30 @@
...
+	class midterm
+		$why_you_do_dis #Global
+		@@no_tests_plz #Class
+		initialize(butthead)
+			@butthead = butthead #Instance and local
+		end
+	end
+	```
 
 1. Write a Monkey class that `eat`s bananas, and `drink`s water
```
Comment (seanculver) [inline quizzes/midterm.md:217]: 9/10 - you should overwrite the eat method. not eat_banana method.

File: quizzes/midterm.md [hunk 23]
```diff
@@ -163,8 +227,18 @@
...
 	```
+	```ruby
+	class Monkey < Animal
+	  def eat_banana
+	    "Om Om Om, oooh aaaah aaaah"
+	  end
+	end
+	```
 
 1. What is the difference between using `extend` and `include`?
```
Comment (seanculver) [inline quizzes/midterm.md:238]: 10/10

File: quizzes/midterm.md [hunk 24]
```diff
@@ -163,8 +227,18 @@
...
+	```ruby
+	#Extend includes the code at the class level while inlcude does in at instance level
+	```
 
 1. Convert this into a case statement
```
Comment (seanculver) [inline quizzes/midterm.md:243]: 10/10

File: quizzes/midterm.md [hunk 25]
```diff
@@ -178,11 +252,31 @@
...
+	```ruby
+	case fruit
+	  when "durrian"
+	    #no thanks
+	  else
+	    #eat
+	end
+	```
 	
 1.  T  /  F Arrays can only hold `String` and `Hash` values?
```
Comment (seanculver) [inline quizzes/midterm.md:264]: 10/10

File: quizzes/midterm.md [hunk 26]
```diff
@@ -178,11 +252,31 @@
...
+	```
+	#False
+	```
 
-1.  What is the scope resolution operator? 
+1.  What is the scope resolution operator?
```
Comment (seanculver) [inline quizzes/midterm.md:269]: 10/10

File: quizzes/midterm.md [hunk 27]
```diff
@@ -178,11 +252,31 @@
...
+	```
+	#::
+	```
 
 1. T / F Ruby expects that you might pass a hash as the last argument to a method.
```
Comment (seanculver) [inline quizzes/midterm.md:274]: 10/10

File: quizzes/midterm.md [hunk 28]
```diff
@@ -178,11 +252,31 @@
...
+	```
+	#True
+	```
 
 1. Extra Credit: What is a Gem?
```
Comment (seanculver) [inline quizzes/midterm.md:279]: 5/10 - not just for ruby but it is a package.

**Comment by seanculver** [comment]
total score 227

**Comment by VeeDoubleYuh** [comment]
Got the last on to #1. Forgot the param when called.

**Pull request closed without merging**